cranelift/codegen/src/isa/aarch64/mod.rs

```diff
@@ -178,6 +178,10 @@ impl TargetIsa for AArch64Backend {
     fn map_regalloc_reg_to_dwarf(&self, reg: Reg) -> Result<u16, systemv::RegisterMappingError> {
         inst::unwind::systemv::map_reg(reg).map(|reg| reg.0)
     }
+
+    fn function_alignment(&self) -> u32 {
+        4
```

Comment:
I'm wondering if this is enough?

Specifically, couldn't we hit the same bug on aarch64, where:
- We have a constant that needs to be 16-byte-aligned (else fault);
- We 16-align it in the function body;
- But the function is only 4-aligned, and has a starting offset != 0 mod 16.

So I think we need to either set `function_alignment` to the highest alignment statically that we know we use, or compute it dynamically based on the alignments actually present in the constant pool, on every architecture; otherwise we don't satisfy the alignment constraints.

Reply:
That's a good point. I'm going to include the minimum alignment requirement in the output when compiling the function, and then take the max of that and the isa function_alignment when emitting instructions.

Reply:
I don't think that this is much of a problem because the only instructions I can think of off the top of my head that have strict alignment requirements are the atomic instructions (or SP-relative operations, but they are not applicable in this case), and it sounds a bit unusual to apply them to literals in the instruction stream. Yes, it is possible to enable alignment checking for almost all relevant instructions, but I am not aware of any platform that does that, and in fact unaligned accesses by default is pretty much an aspect of the ABI at this point.

However, stricter function alignment is in fact recommended for performance reasons, i.e. to maximize fetch bandwidth. The recommendation for some of Arm's recent processors is 32 bytes.

Reply:
@akirilov-arm is it worth changing this to `32` here, or should we leave that for another PR?

Reply:
Doing it in this PR is fine, but you should add a comment that the value has been chosen for performance reasons; otherwise the correctness requirement is 4 bytes, as in your current code.

Reply:
I changed this to `32` and added a comment, thanks!